### ISSUE TITLE
Quick question

### ISSUE THREAD

**Issue by mhv75** [xan-m/Infini-iOS#4]
Tinkering with on iOS side of atcwatch for p8 and other cheap smart watches.

For iOS a lot of those smart watches control music with HID (volume up and down, play pause button presses). Would have to be implemented in Infinitime maybe but a lot easier

**Comment by mhv75** [comment]
Great work btw. Always better to do than to comment ;-)

**Comment by xan-m** [comment]
Yep, that's the boggle here. I posted about this in the PineTime IRC/Matrix/Discord/Telegram Super-Bridged Master Chat a few days ago. I came to the conclusion that, for now, it makes more sense to cater to Apple Music and fully satisfy a hair less than half of iOS users with a tolerable solution than to dump a shitty hack on everyone.

InfiniTime implements music control in a non-standard way, so as it is now, I'd have to make a super-hacky HID device that only makes use of the buttons in the music app and doesn't report any of the now-playing info. Honestly I'd bet even if I did implement it that way, most people wouldn't realize that it was implemented and would leave the music app alone on the PineTime.

In other words, I can spend an afternoon giving a non-zero number of users moderate functionality that I have no issue deleting when a better solution comes along, _*OR*_ I can spend a week's worth of free time hand-making an ad hoc HID interpreter that would barely work for anyone and would still need to be replaced when the better solution is implemented.

I'm looking forward to some of the BT connection issues/secure pairing with the PineTime being ironed out in the not-too-distant future, and then trying to help implement the AMS (apple music service) to control music and update the PineTime music display.

Thanks btw! I'm always happy to collaborate, so let me know if there's anything I can help you with!

**Comment by mhv75** [comment]
Yeah, I understand completely. I figured there must be a reason that big brands like huami still choose to implement HID. They combine it with loading now playing info

**Comment by xan-m** [comment]
I think AMS is the only way to implement this.  With the recent progress on InfiniTime BLE issues, secure bonding is likely on the horizon, which opens the door for AMS and ANCS (for notifications). Marking this issue as closed.
